e2e-tests.Dockerfile

```diff
@@ -59,19 +60,23 @@ RUN tinygo version;   \
     node --version;   \
     swift --version;
 
-RUN wget https://github.com/fermyon/spin/releases/download/${SPIN_VERSION}/spin-${SPIN_VERSION}-linux-amd64.tar.gz &&         \
-    tar -xvf spin-${SPIN_VERSION}-linux-amd64.tar.gz &&                                                                       \
+RUN if [ "${FETCH_SPIN}" = "true" ]; then                                                                                       \
```

Comment:
That would be a further optimisation, but the one thing I wanted to avoid in this PR was introducing new failures.  This was about getting a failing action to pass and unblock PRs; @rajatjindal already has a broader plan that will optimise unnecessary stuff away anyway.